functions/geometry_git.zsh

```diff
@@ -0,0 +1,118 @@
+# geometry_git - please see the readme for documentation on all features
+
+(( $+commands[git] )) || return
+
+_stashes() {
+  git rev-parse --quiet --verify refs/stash >/dev/null \
+    && ansi $GEOMETRY_GIT_COLOR_STASHES $GEOMETRY_GIT_SYMBOL_STASHES
+}
+
+_time() {
+  local last_commit=$(git log -1 --pretty=format:'%at' 2> /dev/null)
+  [[ -z "$last_commit" ]] && ansi $GEOMETRY_COLOR_NO_TIME $GEOMETRY_GIT_NO_COMMITS_MESSAGE && return
+
+  now=$(date +%s)
+  seconds_since_last_commit=$((now - last_commit))
+  geometry::time $seconds_since_last_commit $GEOMETRY_GIT_TIME_DETAILED
+}
+
+_branch() {
+  ansi $GEOMETRY_GIT_COLOR_BRANCH $(git symbolic-ref --short HEAD || git rev-parse --short HEAD)
+}
+
+_status() {
+  _status=$([[ -z "$(git status --porcelain --ignore-submodules HEAD)" ]] && [[ -z "$(git ls-files --others --modified --exclude-standard)" ]] && echo CLEAN || echo DIRTY)
+  ansi ${(e):-\$GEOMETRY_GIT_COLOR_${_status}} ${(e):-\$GEOMETRY_GIT_SYMBOL_${_status}}
+}
+
+_rebase() {
+  git_dir=$(git rev-parse --git-dir)
+  [[ -d "$git_dir/rebase-merge" ]] || [[ -d "$git_dir/rebase-apply" ]] || return
+  echo "$GEOMETRY_GIT_SYMBOL_REBASE"
+}
+
+_remote() {
+  local_commit=$(git rev-parse "@" 2>/dev/null)
+  remote_commit=$(git rev-parse "@{u}" 2>/dev/null)
+
+  [[ $local_commit == "@" || $local_commit == $remote_commit ]] && return
+
+  common_base=$(git merge-base "@" "@{u}" 2>/dev/null) # last common commit
+  [[ $common_base == $remote_commit ]] && echo $GEOMETRY_GIT_SYMBOL_UNPUSHED && return
+  [[ $common_base == $local_commit ]]  && echo $GEOMETRY_GIT_SYMBOL_UNPULLED && return
+
+  echo "$GEOMETRY_GIT_SYMBOL_UNPUSHED $GEOMETRY_GIT_SYMBOL_UNPULLED"
+}
+
+_symbol() { echo ${(j: :):-$(_rebase) $(_remote)} }
+
+_conflicts() {
+  conflicts=$(git diff --name-only --diff-filter=U)
+
+  [[ -n "$conflicts" ]] && return
```

Comment:
This should be `-z` otherwise it will always show the conflict indicator when there are none.